ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/TransactionsReader.scala

```diff
@@ -29,8 +30,28 @@ private[dao] final class TransactionsReader(
     dispatcher: DbDispatcher,
     executionContext: ExecutionContext,
     pageSize: Int,
+    metrics: Metrics,
 ) {
 
+  // Metrics names
+  private val getFlatTransactions: String = "get_flat_transactions"
+  private val lookupFlatTransactionById: String = "lookup_flat_transaction_by_id"
+  private val getTransactionTrees: String = "get_transaction_trees"
+  private val lookupTransactionTreeById: String = "lookup_transaction_tree_by_id"
+  private val getActiveContracts: String = "get_active_contracts"
+
+  // Timers for deserialization on per-query basis
+  private val getFlatTransactionsDeserializationTimer =
+    metrics.daml.index.db.deserialization(getFlatTransactions)
+  private val lookupFlatTransactionByIdDeserializationTimer =
+    metrics.daml.index.db.deserialization(lookupFlatTransactionById)
+  private val getTransactionTreesDeserializationTimer =
+    metrics.daml.index.db.deserialization(getTransactionTrees)
+  private val lookupTransactionTreeByIdDeserializationTimer =
+    metrics.daml.index.db.deserialization(lookupTransactionTreeById)
+  private val getActiveContractsDeserializationTimer =
+    metrics.daml.index.db.deserialization(getActiveContracts)
```

Comment:
I think these could potentially be moved to the `Metrics` class.

Reply:
Good point. Ideally I would also want to pass in the timer as a parameter to `DbDispatcher#runSQL`. Do you think I can tackle it in one go as a separate contribution?

Reply:
Fine with me.

Reply:
Addressed in https://github.com/digital-asset/daml/pull/5927